packages/expo-router/src/modal/Modal.tsx

```diff
@@ -0,0 +1,145 @@
+'use client';
+
+import { type NavigationProp, type ParamListBase } from '@react-navigation/native';
+import { nanoid } from 'nanoid/non-secure';
+import { useEffect, useState } from 'react';
+import { ViewProps } from 'react-native';
+import { type ScreenProps } from 'react-native-screens';
+
+import { useModalContext, type ModalConfig } from './ModalContext';
+import { useNavigation } from '../useNavigation';
+
+export interface ModalProps extends ViewProps {
+  /**
+   * The content of the modal.
+   */
+  children: React.ReactNode;
```

Comment:
```suggestion
  children?: React.ReactNode;
```

Should be optional imo to reduce errors while typing.